NFs with a "." in the name do not compile

If a nimbleFunction, or an RC function, has a "." in the name, then they don't compile.

The errors look different (between NFs and RC functions), but I imagine it's the same root issues.  An example of each is below.

```
library(nimble)

my.RCfun <- nimbleFunction(
    name = 'my.RCfun',
    run = function() { a <- 1 }
)

Cfunction <- compileNimble(my.RCfun, showCompilerOutput = TRUE)

## compiling... this may take a minute. On some systems there may be some compiler warnings that can be safely ignored.
## make: *** No rule to make target `P_9_my.RCfun.o', needed by `P_9_my.RCfun_08_04_09_07_41.so'.  Stop.
## Error: Failed to create the shared library

my.NF <- nimbleFunction(
    name = 'my.NF',
    setup = function() { a <- 0 },
    run = function() { b <- a }
)

Rfun <- my.NF()

Cfunction <- compileNimble(Rfun, showCompilerOutput = TRUE)

## compiling... this may take a minute. On some systems there may be some compiler warnings that can be safely ignored.
## clang++  -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG -DR_NO_REMAP -I  -DEIGEN_MPL2_ONLY=1 -I"/Library/Frameworks/R.framework/Versions/3.4/Resources/library/nimble/include" -Wno-misleading-indentation -Wno-ignored-attributes -Wno-deprecated-declarations  -I/usr/local/include   -fPIC  -Wall -g -O2  -c P_10_my_dot_NF.cpp -o P_10_my_dot_NF.o
## warning: unknown warning option '-Wno-misleading-indentation'; did you mean '-Wno-missing-declarations'? [-Wunknown-warning-option]
## In file included from P_10_my_dot_NF.cpp:10:
## ./P_10_my_dot_NF.h:11:9: error: cannot use dot operator on a type
## class my.NF : public NamedObjects {
##         ^
## ./P_10_my_dot_NF.h:18:24: error: expected ';' after top level declarator
## extern "C" SEXP  new_my.NF (  );
##                        ^
##                        ;
## P_10_my_dot_NF.cpp:13:7: error: variable has incomplete type 'void'
## void  my.NF::run (  )  {
##       ^
## P_10_my_dot_NF.cpp:13:9: error: expected ';' after top level declarator
## void  my.NF::run (  )  {
##         ^
##         ;
## 1 warning and 4 errors generated.
## make: *** [P_10_my_dot_NF.o] Error 1
## Error: Failed to create the shared library
```

--- a/packages/nimble/R/cppDefs_cppProject.R
+++ b/packages/nimble/R/cppDefs_cppProject.R
@@ -240,6 +240,7 @@ cppProjectClass <- setRefClass('cppProjectClass',
                                    },
                                    compileFile = function(names, showCompilerOutput = nimbleOptions('showCompilerOutput'),
                                                           .useLib = UseLibraryMakevars) {
+                                       names <- Rname2CppName(names)
                                        isWindows = (.Platform$OS.type == "windows")
 
                                        includes <- character()

